apps/fhir/bluebutton/views/search.py

```diff
@@ -42,11 +44,29 @@ def search(request, resource_type, *args, **kwargs):
 
     crosswalk = get_crosswalk(request.resource_owner)
 
+    # Get parameters required to replay request for relative links
+    replay_parameters = {}
+
     # If the user isn't matched to a backend ID, they have no permissions
     if crosswalk is None:
         logger.info('Crosswalk for %s does not exist' % request.user)
         return build_error_response(403, 'No access information was found for the authenticated user')
 
+    # Verify paging inputs. Casting an invalid int will throw a ValueError
+    try:
+        start_index = int(request.GET.get(START_PARAMETER, 0))
+        if start_index < 0:
+            raise ValueError
+    except ValueError:
+        return build_error_response(400, '%s must be an integer between zero and the number of results' % START_PARAMETER)
```

Comment:
This can be simplified down to a single `if` statement, or a single `raise` (assuming there's an error handler to transform the exception into an http response)

Reply:
I'm not sure how that's possible. The key thing here is that the int function will raise a ValueError exception on a bad input, and this is the only way I found to consolidate down to one error return call.

Reply:
This is currently equivalent to
```python
start_index = int(request.GET.get(START_PARAMETER, 0))
if start_index < 0:
    return build_error_response(400, '%s must be an integer between zero and the number of results' % START_PARAMETER)
```

Reply:
It's not. The int operation needs to be in a try block because it throws a ValueError exception on anything other than an int. So, if someone passes in startIndex=abcd, you'll get an Exception